src/Components/Promises/PromiseContainer/index.js

```diff
@@ -0,0 +1,60 @@
+/* @flow */
+import React from 'react';
+import { connect } from 'react-redux';
+import type { Connector } from 'react-redux';
+import { fetchPromises } from '../../../Stores/Promises/actions';
+import { promisesLoaded } from '../../../Stores/Promises/selectors';
+import type { Dispatch, State, PromiseType } from '../../../types';
+import PromiseTable from '../PromiseTable';
+
+type OwnProps = {};
+
+type ReduxProps = {
+  promises: ?Array<PromiseType>,
+  hasLoadedPromises: boolean,
+  error: ?string,
+  fetchPromises: () => void,
+};
+
+type Props = OwnProps & ReduxProps;
+
+export class PromiseContainer extends React.Component {
+  props: Props;
+
+  componentDidMount() {
+    const { hasLoadedPromises } = this.props;
+    if (!hasLoadedPromises) {
+      this.props.fetchPromises();
+    }
+  }
+
+  render() {
+    const { promises } = this.props;
+    if (!promises) return null;
+
+    return <PromiseTable promises={promises} />;
+  }
+}
+
+const mapStateToProps = (state: State) => {
+  return {
+    error: state.promises.error,
+    hasLoadedPromises: promisesLoaded(state),
+    promises: state.promises.promises,
+  };
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => {
```

Comment:
I guess it depends on what kind of granularity you want :)

I have usually been doing this:

```
const mapDispatchToProps = dispatch => bindActionCreators({
    ...actions1,
    ...actions2,
}, dispatch);
```

That just passed all actions in those 2 files into the container as props. But here you dont really see what is being pass down, you will have to look at the file for that.

However, if you ever add a new action in either of those files, you dont have to touch the container at all (if you are spreading the props to the component) and simply call the function from the component right away.